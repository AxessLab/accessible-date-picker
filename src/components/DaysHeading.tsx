import React from "react";
import { createUseStyles, useTheme } from 'react-jss';
import { IDatePickerTheme } from '../container/DatePicker';

const useStyles = createUseStyles((theme: IDatePickerTheme) => ({
    abbrDays: {
        border: "none",
        textDecoration: "none",
        padding: theme.spacing[2],
        color: theme.palette.tertiary,
    }
}));

const DaysHeading: React.FC = (props) => {
    const weekdaysAbbr = ['Sunday', 'Monday', 'Tueday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
    const theme = useTheme();
    const styles = useStyles({ ...props, theme });

    return (
        <thead role='presentation'>
            <tr role='presentation'>
                {weekdaysAbbr.map((day, i) => (
                    <th key={i} scope='col' role='presentation'>
                        <abbr className={styles.abbrDays} key={i} aria-hidden='true' title={day}>
                            {day.slice(0, 3)}
                        </abbr>
                    </th>
                ))}
            </tr>
        </thead>
    );
};

export default DaysHeading;