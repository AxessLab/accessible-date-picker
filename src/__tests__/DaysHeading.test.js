import React from "react";
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import DaysHeading from "../components/DaysHeading";

configure({ adapter: new Adapter() });

test("matches snapshot of DaysHeading", () => {
  const wrapper = shallow(<DaysHeading />);
  expect(toJson(wrapper)).toMatchSnapshot();
});