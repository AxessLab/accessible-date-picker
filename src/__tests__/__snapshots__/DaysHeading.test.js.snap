// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of DaysHeading 1`] = `
<thead
  role="presentation"
>
  <tr
    role="presentation"
  >
    <th
      key="0"
      role="presentation"
      scope="col"
    >
      <abbr
        aria-hidden="true"
        key="0"
        title="Sunday"
      >
        Sun
      </abbr>
    </th>
    <th
      key="1"
      role="presentation"
      scope="col"
    >
      <abbr
        aria-hidden="true"
        key="1"
        title="Monday"
      >
        Mon
      </abbr>
    </th>
    <th
      key="2"
      role="presentation"
      scope="col"
    >
      <abbr
        aria-hidden="true"
        key="2"
        title="Tueday"
      >
        Tue
      </abbr>
    </th>
    <th
      key="3"
      role="presentation"
      scope="col"
    >
      <abbr
        aria-hidden="true"
        key="3"
        title="Wednesday"
      >
        Wed
      </abbr>
    </th>
    <th
      key="4"
      role="presentation"
      scope="col"
    >
      <abbr
        aria-hidden="true"
        key="4"
        title="Thursday"
      >
        Thu
      </abbr>
    </th>
    <th
      key="5"
      role="presentation"
      scope="col"
    >
      <abbr
        aria-hidden="true"
        key="5"
        title="Friday"
      >
        Fri
      </abbr>
    </th>
    <th
      key="6"
      role="presentation"
      scope="col"
    >
      <abbr
        aria-hidden="true"
        key="6"
        title="Saturday"
      >
        Sat
      </abbr>
    </th>
  </tr>
</thead>
`;
